elastic/kibana
# [Infrastructure UI] Asset Details: Add pins to the metadata table

Closes #155190

## Summary

This PR adds the possibility to pin different rows inside the metadata table in asset details embeddable. The pins are persisted in the local storage and should be available after refreshing/reopening the host flyout. The order and sorting are explained in [this comment](https://github.com/elastic/kibana/issues/155190#issuecomment-1523335704), so basically we keep the original sorting order of the table (`host`, `cloud`, `agent`) also for the pins.

## Testing 
- Go to hosts view and open a single host flyout (metadata tab)
- Try to add / remove pins
- Check if the pins are persisted after a page refresh


https://github.com/elastic/kibana/assets/14139027/62873e7e-b5f0-444c-94ff-5e19f2f46f58
